scipy/ndimage/filters.py

```diff
@@ -1186,3 +1186,102 @@ def generic_filter(input, function, size=None, footprint=None,
     _nd_image.generic_filter(input, function, footprint, output, mode,
                          cval, origins, extra_arguments, extra_keywords)
     return return_value
+
+
+@docfiller
+def line_integral_convolution(input, velocity, kernel, origin=0, order=3, reweight=True):
+    """
+    Line integral convolution of an input image with an arbitrary kernel.
+    Lines are defined by a velocity field.
+
+    Parameters
+    ----------
+    input : array_like
+        Random image to be convolved
+        (can be created for instance with
+        "1. * (np.random.random(shape) > 0.5)")
+    velocity: array_like
+        One velocity vector for each pixel.
+        Must have shape "input.shape + (input.ndim,)"
+        First dimensions are identical to the shape of the random input
+        image, the last dimension defines the coordinate of the velocity.
+    kernel: array_like
+        1-D array with at least on element. Defines the convolution kernel
+        (e.g. a Gaussian kernel constructed with
+        scipy.stats.norm.norm.pdf(np.linspace(-3,3,50)))
+        The flow direction can be visualized by using an asymmetric kernel
+        in combination with an "origin" parameter equal to -len(kernel) // 2
+    origin: int, optional
+        Placement of the filter, by default 0 (which is correct for
+        symmetric kernels). The flow direction can be visualized by using an
+        asymmetric kernel in combination with an "origin" parameter equal to
+        -len(kernel) // 2
+    order: int, optional
+        The order of the spline interpolation used for interpolating the input
+        image and the velocity field, by default 3.
+        See the documentation of
+        "scipy.ndimage.interpolation.map_coordinates" for more details
+    reweight: bool, optional
+        Element-wise reweighting of the filter weights to correct for boundary
+        effects, by default True
+        If set to False, the output image has lower values at the boundary
```

Comment:
A scipy docstring needs to have a ``Returns`` section. ``See Also``, ``References`` and ``Examples`` are also desired - an example with a plot would be nice here.